docs/root/configuration/http/http_filters/basic_auth_filter.rst

```diff
@@ -26,13 +26,49 @@ An example configuration of the filter may look like the following:
 
 .. code-block:: yaml
 
-  users:
-    inline_string: |-
-        user1:{SHA}hashed_user1_password
-        user2:{SHA}hashed_user2_password
+  http_filters:
+  - name: envoy.filters.http.basic_auth
+    typed_config:
+      "@type": type.googleapis.com/envoy.extensions.filters.http.basic_auth.v3.BasicAuth
+      users:
+        inline_string: |-
+          user1:{SHA}hashed_user1_password
+          user2:{SHA}hashed_user2_password
 
 Note that only SHA format is currently supported. Other formats may be added in the future.
 
+Per-Route Configuration
+-----------------------
+
+An example configuration of the route filter may look like the following:
+
+.. code-block:: yaml
+
+  route_config:
+    name: local_route
+    virtual_hosts:
+    - name: local_service
+      domains: ["*"]
+      routes:
+      - match: { path: "/admin" }
+        route: { cluster: some_service }
+        typed_per_filter_config:
+          envoy.filters.http.basic_auth:
+            "@type": type.googleapis.com/envoy.extensions.filters.http.basic_auth.v3.BasicAuthPerRoute
+            users:
+              inline_string: |-
+                admin:{SHA}0DPiKuNIrrVmD8IUCuw1hQxNqZc=
```

Comment:
Nit: could we use the same masked password as the previous example?

```suggestion
                admin:{SHA}hashed_admin_password

```

Reply:
Done